carbonate/sync.py

```diff
@@ -14,31 +14,38 @@
 
 def sync_from_remote(sync_file, remote, staging, rsync_options):
     try:
-        try:
-            os.makedirs(os.path.dirname(staging))
-        except OSError:
-            pass
+        os.makedirs(os.path.dirname(staging))
+    except OSError:
+        pass
+
+    cmd = " ".join(['rsync', rsync_options, '--files-from',
+                    sync_file.name, remote, staging + '/'
+                    ])
 
-        cmd = " ".join(['rsync', rsync_options, '--files-from',
-                        sync_file.name, remote, staging
-                        ])
+    print("  - Rsyncing metrics: %s" % cmd)
 
-        print("  - Rsyncing metrics")
+    proc = subprocess.Popen(cmd,
+                            shell=True,
+                            universal_newlines=True,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.STDOUT)
 
-        proc = subprocess.Popen(cmd,
-                                shell=True,
-                                stdout=subprocess.PIPE,
-                                stderr=subprocess.STDOUT)
+    for line in iter(proc.stdout.readline, b''):
+        sys.stdout.write(line.decode("utf-8"))
+        sys.stdout.flush()
 
-        for line in iter(proc.stdout.readline, b''):
-            sys.stdout.write(line.decode("utf-8"))
-            sys.stdout.flush()
-    except subprocess.CalledProcessError as e:
-        logging.warn("Failed to sync from %s! %s" % (remote, e))
+    proc.communicate()
+    print("    rc: %d" % proc.returncode)
+
+    if proc.returncode != 0:
+        logging.warn("Failed to sync from %s! rsync rc=%d",
+                        remote, proc.returncode)
```

Comment:
Almost. Lint is not happy - 
carbonate/sync.py:42:25: E127 continuation line over-indented for visual indent